docs/notebooks/data_transformation.pct.py

```diff
@@ -0,0 +1,227 @@
+
+# %%
+num_initial_points = 50
+
+observer = mk_observer(function)
+
+initial_query_points = search_space.sample_sobol(num_initial_points)
+initial_data = observer(initial_query_points)
+
+
+# %% [markdown]
+# ## Model the objective function
+#
+# The Bayesian optimization procedure estimates the next best points to query by using a probabilistic model of the objective. We'll use a Gaussian process (GP) model, built using GPflow. We also set priors over the hyperparameters.
+
+# %%
+def build_gp_model(data):
+    variance = tf.math.reduce_variance(data.observations)
+    kernel = gpflow.kernels.Matern52(
+        variance=variance, lengthscales=[0.2] * data.query_points.shape[-1]
+    )
+    prior_scale = tf.cast(1.0, dtype=tf.float64)
+    kernel.variance.prior = tfp.distributions.LogNormal(
+        tf.cast(-2.0, dtype=tf.float64), prior_scale
+    )
+    kernel.lengthscales.prior = tfp.distributions.LogNormal(
+        tf.math.log(kernel.lengthscales), prior_scale
+    )
+    gpr = gpflow.models.GPR(
+        data.astuple(), kernel, mean_function=gpflow.mean_functions.Constant(), noise_variance=1e-5
+    )
+    gpflow.set_trainable(gpr.likelihood, False)
+
+    return GaussianProcessRegression(
+        model=gpr,
+        optimizer=Optimizer(
+            gpflow.optimizers.Scipy(), minimize_args={"options": dict(maxiter=100)}
+        ),
+        num_kernel_samples=100,
+    )
+
+
+# build the model
+model = build_gp_model(initial_data)
+
+
+# %% [markdown]
+# ## Run the optimization loop
+#
+# We can now run the Bayesian optimization loop by defining a `BayesianOptimizer` and calling its `optimize` method.
+#
+# The optimizer uses an acquisition rule to choose where in the search space to try on each optimization step. We'll be using Expected improvement acquisition function - it is used by default, so no need to specify it.
+#
+# We'll run the optimizer for 100 steps. Note: this may take a while!
+
+# %%
+num_acquisitions = 100
+
+bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+result = bo.optimize(num_acquisitions, initial_data, model)
+dataset = result.try_get_final_dataset()
+
+
+# %% [markdown]
+# ## Explore the results
+#
+# We can now get the best point found by the optimizer. Note this isn't necessarily the point that was last evaluated. We will also plot regret for each optimization step.
+#
+# We can see that the optimization did not get close to the global optimum of -210.
+
+# %%
+query_points = dataset.query_points.numpy()
+observations = dataset.observations.numpy()
+
+arg_min_idx = tf.squeeze(tf.argmin(observations, axis=0))
+
+print(f"query point: {query_points[arg_min_idx, :]}")
+print(f"observation: {observations[arg_min_idx, :]}")
+
+
+# %% [markdown]
+# Once ask-tell optimization is over, you can extract an optimization result object and perform whatever analysis you need, just like with regular Trieste optimization interface. For instance, here we will plot regret for each optimization step.
```

Comment:
This section (and code below) needs to be updated. It came straight from ask tell notebook, right?